python/pip_install/extract_wheels/bazel_test.py

```diff
@@ -1,11 +1,80 @@
+import os
+import shutil
 import unittest
+import unittest.mock
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
 
-from python.pip_install.extract_wheels.bazel import generate_entry_point_contents
+from python.pip_install.extract_wheels import annotation
+from python.pip_install.extract_wheels.bazel import (
+    extract_wheel,
+    generate_entry_point_contents,
+)
+
+
+class MockWheelInstance:
+    """A mock for python.pip_install.extract_wheels.wheel.Wheel."""
+
+    def __init__(
+        self,
+        name: str,
+        version: str,
+        path: str,
+        dependencies: List[str],
+        entry_points: Dict[str, Tuple[str, str]],
+    ):
+        self.name = name
+        self.path = path
+        self.version = version
+        self.dependency_list = dependencies
+        self.entry_point_dict = entry_points
+
+    def dependencies(self, extras_requested):
+        # Since the caller can customize the dependency list to their liking,
+        # we don't need to act on the extras_requested.
+        return set(self.dependency_list)
+
+    def entry_points(self):
+        return self.entry_point_dict
+
+    def unzip(self, directory):
+        # We don't care about actually generating any files for our purposes.
+        pass
+
+
+def parse_starlark(filename: os.PathLike) -> Dict[str, unittest.mock.MagicMock]:
+    """Parses a Starlark file.
+
+    Args:
+        filename: The name of the file to parse as Starlark.
+
+    Returns:
+        A dictionary of MagicMock instances for each of the functions that was
+        invoked in the Starlark file.
+    """
+    starlark_globals = {
+        "filegroup": unittest.mock.MagicMock(),
+        "glob": unittest.mock.MagicMock(return_value=["<glob()>"]),
+        "load": unittest.mock.MagicMock(),
+        "package": unittest.mock.MagicMock(),
+        "py_binary": unittest.mock.MagicMock(),
+        "py_library": unittest.mock.MagicMock(),
+    }
+    compiled_starlark = compile(Path(filename).read_text(), filename, "exec")
+    eval(compiled_starlark, starlark_globals)
+    return starlark_globals
 
 
 class BazelTestCase(unittest.TestCase):
+    def setUp(self):
+        self.tmpdir = Path(os.environ["TEST_TMPDIR"]) / "tmpdir"
+        self.tmpdir.mkdir()
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdir)
+
     def test_generate_entry_point_contents(self):
-        got = generate_entry_point_contents("sphinx.cmd.build:main")
+        got = generate_entry_point_contents("sphinx.cmd.build", "main")
```

Comment:
Since this file was missing the dunder main check to invoke `unittest.main()`, I had to fix these tests up too.